zipline/pipeline/factors/technical.py

```diff
@@ -144,6 +159,253 @@ def compute(self, today, assets, out, close, volume):
+        2017-03-16       0    -.02    .01
+        2017-03-17     .01     .04   -.01
+        2017-03-20     .02    -.03   -.02
+        2017-03-21     .03     .01   -.02
+        2017-03-22     .04    -.02   -.02
+
+    Suppose we are interested in SPY's rolling returns correlation with each
+    stock from 2017-03-17 to 2017-03-22, using a 5-day look back window (that
+    is, we calculate each correlation coefficient over 5 days of data). We can
+    achieve this by doing::
+
+        rolling_correlations = RollingPearsonOfReturns(
+            baseline_asset=Equity(8554),
+            returns_length=10,
+            correlation_length=5,
+        )
+
+    The result of computing ``rolling_correlations`` from 2017-03-17 to
+    2017-03-22 gives::
+
+                       SPY   MSFT     FB
+        2017-03-17       1    .15   -.96
+        2017-03-20       1    .10   -.96
+        2017-03-21       1   -.16   -.94
+        2017-03-22       1   -.16   -.85
+
+    Note that the column for SPY is all 1's, as the correlation of any data
+    series with itself is always 1. To understand how each of the other values
+    were calculated, take for example the .15 in MSFT's column. This is the
+    correlation coefficient between SPY's returns looking back from 2017-03-17
+    (-.03, -.02, -.01, 0, .01) and MSFT's returns (.03, -.03, .02, -.02, .04).
+
+    See Also
+    --------
+    :class:`zipline.pipeline.factors.technical.RollingReturnsRegression`
+    """
+    def compute(self, today, assets, out, data, baseline_asset):
+        asset_col = searchsorted(list(assets), baseline_asset.sid)
+        out[:] = corrcoef(data, rowvar=0)[asset_col]
+
+
+class RollingSpearmanOfReturns(_RollingCorrelationOfReturns):
+    """
+    Calculates the Spearman rank correlation coefficient of the returns of the
+    given asset with the returns of all other assets.
+
+    Parameters
+    ----------
+    baseline_asset : zipline.assets.Asset
+        The asset to correlate with all other assets.
+    returns_length : int > 1
+        Length of the lookback window over which to compute returns.
+    correlation_length : int > 1
+        Length of the lookback window over which to compute each correlation
+        coefficient.
+
+    See Also
+    --------
+    :class:`zipline.pipeline.factors.technical.RollingSpearmanOfReturns`
+    """
+    def compute(self, today, assets, out, data, baseline_asset):
+        asset_col = searchsorted(list(assets), baseline_asset.sid)
+        out[:] = spearmanr(data)[0][asset_col]
+
+
+class RollingLinearRegressionOfReturns(CustomFactor, SingleInputMixin):
+    """
+    Perform an ordinary least-squares regression predicting the returns of all
+    other assets on the given asset.
+
+    Parameters
+    ----------
+    baseline_asset : zipline.assets.Asset
+        The asset to regress against all other assets.
+    returns_length : int > 1
+        Length of the lookback window over which to compute returns.
```

Comment:
This (and other notes like it) could probably do with another sentence noting that "one day" returns actually require a window_length of 2.